lib/rspec/support/object_inspector.rb

```diff
@@ -3,23 +3,51 @@ module Support
     # Provide additional output details beyond what `inspect` provides when
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
+      DEFAULT_OPTIONS = {
+        :nested => false,
+        :joiner => ',',
+        :braces => true,
+        :trailing_comma => false
+      }
       # @api private
-      def self.inspect(object)
-        if Time === object
+      # rubocop:disable CyclomaticComplexity
+      # rubocop:disable MethodLength
+      def self.inspect(object, options=DEFAULT_OPTIONS)
+        case object
+        when Time
           format_time(object)
-        elsif defined?(DateTime) && DateTime === object
-          format_date_time(object)
-        elsif defined?(BigDecimal) && BigDecimal === object
-          "#{object.to_s 'F'} (#{object.inspect})"
-        elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
-          object.description
+        when Hash
+          format_hash(object, options)
         else
-          registered_klasses.each do |klass, inspector|
-            return inspector.call(object) if klass === object
+          if defined?(DateTime) && DateTime === object
+            format_date_time(object)
+          elsif defined?(BigDecimal) && BigDecimal === object
+            "#{object.to_s 'F'} (#{object.inspect})"
+          elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
+            object.description
+          else
+            registered_klasses.each do |klass, inspector|
+              return inspector.call(object) if klass === object
+            end
+            object.inspect
           end
+        end
+      end
+      # rubocop:enable CyclomaticComplexity
+      # rubocop:enable MethodLength
 
-          object.inspect
+      # @private
+      def self.format_hash(object, options)
+        nested_options = { :braces => true, :joiner => ', ' }
+
+        pairs = object.sort_by { |k, _| k.to_s }.map do |key, value|
+          "#{inspect(key, nested_options)} => #{inspect(value, nested_options)}"
         end
+
+        results = pairs.join(options[:joiner])
+        results = "{#{results}}" if options[:braces]
+        results += ',' if options[:trailing_comma]
```

Comment:
I'd prefer you used `results << ',' if ...` here.  `+=` allocates a new string object where as `<<` doesn't, so the latter is going to perform better.